lang/cs/Org.Apache.REEF.Client/Common/HttpClientHelper.cs

```diff
@@ -99,6 +105,11 @@ internal static string GetAppId(string filePath)
             }
         }
 
+        /// <summary>
+        /// Get and set application state
+        /// </summary>
+        public ApplicationState AppStatus { get; set; }
```

Comment:
Why is the `public` and mutable? That would allow anyone with a handle on this object to make arbitrary changes.